src/Tools/ExternalAccess/AspNetCore/EmbeddedLanguages/AspNetCoreCSharpRouteSyntaxDetector.cs

```diff
@@ -2,37 +2,41 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-using System;
 using System.Collections.Generic;
 using System.Collections.Immutable;
 using System.Diagnostics.CodeAnalysis;
-using System.Text;
 using System.Threading;
 using Microsoft.CodeAnalysis.CSharp.EmbeddedLanguages.LanguageServices;
 using Microsoft.CodeAnalysis.EmbeddedLanguages;
 
-namespace Microsoft.CodeAnalysis.ExternalAccess.AspNetCore.EmbeddedLanguages
+namespace Microsoft.CodeAnalysis.ExternalAccess.AspNetCore.EmbeddedLanguages;
+
+internal sealed class AspNetCoreCSharpRouteSyntaxDetector
 {
-    internal sealed class AspNetCoreCSharpRouteSyntaxDetector
-    {
-        public static readonly AspNetCoreCSharpRouteSyntaxDetector Instance = new();
+    public static readonly AspNetCoreCSharpRouteSyntaxDetector Instance = new();
 
-        private readonly EmbeddedLanguageDetector _detector = new(
+    private static readonly EmbeddedLanguageDetector s_detector;
+    
+    static AspNetCoreCSharpRouteSyntaxDetector()
```

Comment:
>static AspNetCoreCSharpRouteSyntaxDetector() [](http://example.com/codeflow?start=4&length=44)

Why the static ctor here and not in the other detectors?

Reply:
they're special in that they get a static singleton unattached to any particular compilation.  We didn't want to tie them tightly to a compilation instance like we have with ours.  This does mean that on any particular token, they may have to recompute some information from teh compilation over and over again.  But so far it hasn't been an issue.

if we run into isues with them, we could change the api shape here and have them move to that.

Reply:
That describes why this doesn't derive from AbstractLanguageDetector, which is good context, but my specific question here was more syntactic in nature: why use the explicit static ctor when you didn't in the other files.

Reply:
oh.  because i had no need to reference the LanguageIdentifiers in any other locations.  So i didn't feel the need to make that into a field.  And, not having it as a field meant i was going to have to repeat it twice.  So i just made a static ctor, made it a local, and referenced that local twice :)